src/cmfd_data.F90

```diff
@@ -254,7 +254,7 @@ subroutine compute_xs()
 
                 ! Set the energy bin if needed
                 if (energy_filters) then
-                  filter_matches(i_filter_ein) % bins % data(1) = ng - h + 1
+                  filter_matches(i_filter_ein) % bins % data(1) = 12*(ng - h) + 1
```

Comment:
Looking at how this is used, I don't think this is the correct fix actually. The order of bins in the tally will be something like:
```
mesh bin 1, xmin out, group 1
mesh bin 1, xmin out, group 2
...
mesh bin 1, xmin out, group N
mesh bin 2, xmin in, group 1
mesh bin 2, xmin in, group 2
...
```
So the way we are using `score_index` below is just fundamentally wrong. It is assuming that `mesh bin i, xmin out, group j` and `mesh bin i, xmin in, group j` are consecutive in memory, which in fact that are actually separated by the number of energy groups.